bool detect
bool lock
geometry_msgs/Point position
geometry_msgs/Point position_best
geometry_msgs/Point normal